internal/udp/conn.go

```diff
@@ -91,14 +78,20 @@ func Listen(network string, laddr *net.UDPAddr) (*Listener, error) {
 	}
 
 	l := &Listener{
-		pConn:     conn,
-		acceptCh:  make(chan *Conn),
-		conns:     make(map[string]*Conn),
-		accepting: true,
-		doneCh:    make(chan struct{}),
+		pConn:    conn,
+		acceptCh: make(chan *Conn),
+		conns:    make(map[string]*Conn),
+		doneCh:   make(chan struct{}),
 	}
+	l.accepting.Store(true)
+	l.connWG.Add(1)
+	l.readWG.Add(1)
 
 	go l.readLoop()
+	go func() {
+		l.connWG.Wait()
+		_ = l.pConn.Close()
```

Comment:
Still reading the code, but this jumps out at me. Anyway we can not ignore this error?

Reply:
It would be better to be returned from the final `Close()` call of `Listener` and `Conn`s. (underlying `Conn` are shared between them.)
https://github.com/pion/dtls/blob/7ee6cc76ae90b3b66c25f9313c21e7c7f725c9ed/internal/udp/conn.go#L58-L61
https://github.com/pion/dtls/blob/7ee6cc76ae90b3b66c25f9313c21e7c7f725c9ed/internal/udp/conn.go#L187-L190

Reply:
Fixed to store this error and return from the final `Close()`.